packages/destructure/src/index.ts

```diff
@@ -106,7 +119,11 @@ export function destructure<T extends ReactiveSource, O extends DestructureOptio
     const calc = getter(key);
     if (config.deep && isReactiveObject(value))
       result[key] = destructure(calc, { ...config, memo });
-    else result[key] = memo ? createMemo(calc, undefined, options) : calc;
+    else
+      result[key] =
+        memo && (!config.normalize || calc.length === 0)
```

Comment:
There is one thing I'm realizing about this.
```ts
const [callback, setCallback] = createSignal((a, b) => a + b)

const {cb} = destructure({get cb() { return callback() }})

setCallback(() => () => 69)

expect(cb()).toBe(69) // fail: still calls the original function
```
Even if we want to avoid `cb()(1, 2)` syntax of normal destructure, maybe it should still support "reactive callbacks", the same `props.callback(1, 2)` would if you passed `callback={someSignal()}` in JSX.
Although Solid doesn't allow for reactive callbacks in `on___` props, so it may be fine to restrict that as well.